docs/docs/building-modules/04-query-services.md

```diff
@@ -54,3 +54,26 @@ Here's an example implementation for the bank module:
 ```go reference
 https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/bank/keeper/grpc_query.go
 ```
+
+### Calling queries from the State Machine
+
+Introducing `module_query_safe` proto annotation which is used to describe a query is a deterministic (means safe to query), according to ADR-033 a module's Keeper can be called from another module's Keeper and CosmWasm contracts can also directly interact with Query Client.
+
+if the `module_query_safe` annotation set to `true` it means:
+ - The query is deterministic, given a block height it will return the same response upon multiple calls (ex: 1000 or more no.of calls), and doesn't introduce any state-machine breaking changes across the SDK patch version.
```

Comment:
```suggestion
 - The query is deterministic: given a block height it will return the same response upon multiple calls, and doesn't introduce any state-machine breaking changes across SDK patch versions.
```